### ISSUE TITLE
create a module with pecan logic

### ISSUE THREAD

**Issue by javisantana** [CartoDB/cartodb#4761]
Pecan logic, the means the decision tree and so on, should be an external module. 

The reason is that we need to render the same maps we are rendering on cartodb on different places to create tests and analysis

cc @javierarce @ohasselblad @andrewxhill

**Comment by andrewxhill** [comment]
i <3 this ticket

**Comment by saleiva** [comment]
me too!  Prioritize!

**Comment by javisantana** [comment]
adding as a milestone objetive

**Comment by ohasselblad** [comment]
does this mean that we'll get something like these?
```
sublayer.setPecanMap(); // set sublayer to 'best' map
var recmaps = sublayer.getPecanMaps(5); // get five 'best' maps
```

**Comment by viddo** [comment]
@ohasselblad no, not for his change at least, sorry
done & merged in #4999
